doc/architecture-quick-reference.rst

```diff
@@ -0,0 +1,444 @@
+
+This is a quick reference guide to IoTempower's architecture and key concepts.
+
+For detailed information, see:
+
+- `Architecture Overview <architecture.rst>`_ - Complete system architecture
+- `Device Architecture <device-architecture.rst>`_ - Device framework details
+- `Deployment Process <deployment-process.rst>`_ - Build and deploy workflow
+
+
+Key Concepts
+------------
+
+Systems, Nodes, and Devices
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block::
+
+   IoT System (e.g., "my-home")
+   └── Nodes (e.g., "living-room", "bedroom")
+       └── Devices (e.g., "temperature", "motion", "led")
+
+**System**: A complete IoT installation (one WiFi network, one MQTT broker)
+
+**Node**: A microcontroller (ESP8266/ESP32) with connected devices
+
+**Device**: A sensor or actuator (temperature sensor, LED, button, etc.)
+
+
+Device Class Hierarchy
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block::
+
+   Device (base class)
+   ├── Input_Base → Input, Analog
+   ├── Output
+   ├── I2C_Device → BMP180, BH1750, etc.
+   ├── Dht11, Dht22, Ds18b20
+   └── RGB_Base → RGB_Single
+
+Each level adds functionality:
+
+1. **Device**: Core lifecycle, MQTT, polling
+2. **Input_Base**: Pin management, buffers
+3. **Input**: Digital reading, debouncing
+
+
+Configuration Files
+-------------------
+
+system.conf (System Level)
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: bash
+
+   # WiFi and network
+   IOTEMPOWER_AP_NAME="my-iot-network"
+   IOTEMPOWER_AP_PASSWORD="secret123"
+   
+   # MQTT broker
+   MQTT_HOST="192.168.1.100"
+   MQTT_USER="mqtt"
+   MQTT_PASSWORD="mqttpass"
+
+node.conf (Node Level)
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: bash
+
+   # Board type
+   board="wemos_d1_mini"
+   # or: board="esp32dev"
+
+setup.cpp (Device Level)
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: cpp
+
+   // Define your devices
+   led("status", D1);
+   button("door", D2).with_pull_up();
+   dht22("climate", D4);
+   
+   // Add interactions
+   door.with_on_change_callback(*new Callback([](Device& dev) {
+       if (dev.is("on")) {
+           status.on();
+       }
+       return true;
+   }));
+
+
+Device Lifecycle
+----------------
+
+Each device goes through these stages:
+
+1. **Construction**
+   
+   .. code-block:: cpp
+   
+      Device(const char* name, unsigned long pollrate_us)
+
+2. **Registration**
+   
+   Automatically added to DeviceManager
+
+3. **Start**
+   
+   .. code-block:: cpp
+   
+      virtual void start() { _started = true; }
+   
+   Initialize hardware (set pin modes, initialize sensors)
+
+4. **Measure**
+   
+   .. code-block:: cpp
+   
+      virtual bool measure() { return true; }
+   
+   Read sensor values (called at pollrate intervals)
+
+5. **Publish**
+   
+   Changed values automatically published via MQTT
+
+6. **Receive**
+   
+   Commands received from MQTT subscriptions
+
+
+MQTT Topics
+-----------
+
+Topic Structure
+~~~~~~~~~~~~~~~
+
+.. code-block::
+
+   <system>/<node>/<device>/[subdevice]/[command]
```

Comment:
The system-name is not used in the actual mqtt topic. So we start at a node (or often at a room in which several nodes are)